src/my-bot.ts

```diff
@@ -1,20 +1,71 @@
+import {Position} from './position';
+import {randBetween} from './random';
+import {BoardState} from './boardState';
+import {ConfigurationGenerator} from './configurationGenerator';
+import {Shot} from './shot';
+import {ShipPlacer} from './shipPlacer';
+
 export class MyBot {
+    static letters:string = 'ABCDEFGHIJ';
+    static shipPlacer = new ShipPlacer([5,4,3,3,2]);
+
     public getShipPositions() {
-        return [
-            { StartingSquare: { Row: "A", Column: 1 }, EndingSquare : { Row: "A", Column: 5 } },
-            { StartingSquare: { Row: "C", Column: 1 }, EndingSquare : { Row: "C", Column: 4 } },
-            { StartingSquare: { Row: "E", Column: 1 }, EndingSquare : { Row: "E", Column: 3 } },
-            { StartingSquare: { Row: "G", Column: 1 }, EndingSquare : { Row: "G", Column: 3 } },
-            { StartingSquare: { Row: "I", Column: 1 }, EndingSquare : { Row: "I", Column: 2 } },
-        ]
+        // return [
+        //     { StartingSquare: { Row: "A", Column: 1 }, EndingSquare : { Row: "A", Column: 5 } },
+        //     { StartingSquare: { Row: "C", Column: 1 }, EndingSquare : { Row: "C", Column: 4 } },
+        //     { StartingSquare: { Row: "E", Column: 1 }, EndingSquare : { Row: "E", Column: 3 } },
+        //     { StartingSquare: { Row: "G", Column: 1 }, EndingSquare : { Row: "G", Column: 3 } },
+        //     { StartingSquare: { Row: "I", Column: 1 }, EndingSquare : { Row: "I", Column: 2 } },
+        // ]
+        //return BoardState.getRandomStartingConfiguration([5,4,3,3,2]);
+        return MyBot.shipPlacer.chooseConfiguration();
     }
 
     public selectTarget(gamestate) {

```

Comment:
Would be nice to type `gamestate ` (especiially since I think you already have a custom type for it)